nilearn/surface/surface.py

```diff
@@ -2085,3 +2086,212 @@ def new_img_like(ref_img, data):
+    uv = np.concatenate([u, v])
+    vu = np.concatenate([v, u])
+
+    return csr_matrix((ee, (uv, vu)), shape=(n, n))
+
+
+def smooth_img(
+    imgs,
+    iterations=1,
+    distance_weights=False,
+    vertex_weights=None,
+    center_surround_knob=0,
+):
+    """Smooth values along the surface.
+
+    Parameters
+    ----------
+    imgs : SurfaceImage
+        The surface whose is to be smoothed.
+        In the case of 2D data, each sample is smoothed independently.
+
+    iterations : :obj:`int`, default = 1
+        The number of times to repeat the smoothing operation
+        (it must be a positive value).
+        Defaults to 1
+
+    distance_weights : :obj:`bool`, default = False
+        Whether to add distance-based weighting to the smoothing.
+        With such weights, the value calculated for each vertex
+        at each iteration is the weighted sum of neighboring vertices
+        where the weight on each neighbor is the inverse
+        of the distances to it.
+
+    vertex_weights : SurfaceImage or None, default = None
+        A SurfaceImage whose data are vector of weights, one per vertex.
+        These weights are normalized and
+        applied to the smoothing
+        after the application of center-surround weights.
+
+    center_surround_knob : :obj:`float`, default = 0
+        The relative weighting of the center and the surround
+        in each iteration of the smoothing.
+        If the value of the knob is `k`,
+        then the total weight of vertices that are neighbors
+        of a given vertex (the vertex's surround)
+        is `2**k` times the weight of the vertex itself (the center).
+        A value of 0 (the default) means that, in each smoothing iteration,
+        each vertex is updated with the average of its value
+        and the average value of its neighbors.
+        A value of `-inf` results in no smoothing because the entire
+        weight is on the center, so each vertex is updated with its own value.
+        A value of `inf` results in each vertex being updated
+        with the average of its neighbors without including its own value.
+
+    Returns
+    -------
+    smoothed_imgs : SurfaceImage
+        SurfaceImage with smoothed data at each vertex.
+
+    Examples
+    --------
+    >>> from nilearn import datasets, surface
+    >>> curv = datasets.load_fsaverage_data(data_type="curvature")
+    >>> curv_smooth = surface.smooth_img(curv, iterations=50)
+
+    """
+    # First, calculate the center and surround weights for the
+    # center-surround knob.
+    center_weight = 1 / (1 + np.exp2(-center_surround_knob))
+    surround_weight = 1 - center_weight
+    if surround_weight == 0:
+        # There's nothing to do in this case.
+        return imgs
+
+    # Calculate the adjacency matrix either weighting
+    # by inverse distance or not weighting (ones)
+    values = "invlen" if distance_weights else "ones"
+
+    _ = _sanitize_weights(imgs, vertex_weights=vertex_weights)
```

Comment:
I extracted the weight setting into a separate function and made the smoothing function not return the weights anymore to be closer to the smoothing function in volume in terms of API.
Also if users REALLY want to access the weights they can use the private function.

That being said this makes it even more clear that weights are not used in the current implementation.